include/net/net_pkt.h

```diff
@@ -2160,6 +2184,228 @@ struct net_pkt *net_pkt_rx_allocate_with_buffer(struct net_if *iface,
+ *
+ * @param pkt The net_pkt which cursor's position is going to be returned
+ *
+ * @return cursor's position
+ */
+static inline void *net_pkt_cursor_get_pos(struct net_pkt *pkt)
+{
+	return pkt->cursor.pos;
+}
+
+/**
+ * @brief Skip some data from a net_pkt
+ *
+ * Note: net_pkt's cursor should be properly initialized
+ *       Cursor will be updated according to parameter.
+ *       Depending on the value of pkt->overwrite bit, this function
+ *       will affect the buffer length or not: if it's 0, skip will
+ *       actually apply the move in the buffer as it had written in it.
+ *
+ * @param pkt    The net_pkt which cursor will be updated to skip given
+ *               amount of data from the buffer.
+ * @param amount Amount of data to skip in the buffer
+ *
+ * @return 0 in success, negative errno code otherwise.
+ */
+int net_pkt_skip(struct net_pkt *pkt, size_t skip);
+
+/**
+ * @brief Memset some data in a net_pkt
+ *
+ * Note: net_pkt's cursor should be properly initialized and,
+ *       eventally, properly positioned using net_pkt_skip.
+ *       Cursor will be updated according to parameter.
+ *
+ * @param pkt    The net_pkt which cursor will be updated to skip given
+ *               amount of data from the buffer.
+ * @param byte   The byte to write in memory
+ * @param amount Amount of data to memset with given byte
+ *
+ * @return 0 in success, negative errno code otherwise.
+ */
+int net_pkt_memset(struct net_pkt *pkt, int byte, size_t amount);
+
+/**
+ * @brief Copy data from a packet into another one.
+ *
+ * Note: Both net_pkt cursors should be properly initialized and,
+ *       eventally, properly positioned using net_pkt_skip.
+ *       Cursors will be updated according to parameters.
+ *
+ * @param pkt_dst Destination network packet.
+ * @param pkt_src Source network packet.
+ * @param length  Length of data to be copied.
+ *
+ * @return 0 on success, negative errno code otherwise.
+ */
+int net_pkt_copy_new(struct net_pkt *pkt_dst,
+		     struct net_pkt *pkt_src,
+		     size_t length);
+
+/**
+ * @brief Clone pkt and its fragment chain.
+ *
+ * @param pkt Original pkt to be cloned
+ * @param timeout Timeout to wait for free buffer
+ *
+ * @return NULL if error, cloned packet otherwise.
+ */
+struct net_pkt *net_pkt_clone_new(struct net_pkt *pkt, s32_t timeout);
+
+/**
+ * @brief Read some data from a net_pkt
+ *
+ * Note: net_pkt's cursor should be properly initialized and,
+ *       eventally, properly positioned using net_pkt_skip.
+ *       Cursor will be updated according to parameters.
+ * @param pkt    The network packet from where to read some data
+ * @param data   The destination buffer where to copy the data
+ * @param length The amount of data to copy
+ *
+ * @return 0 on success, negative errno code otherwise.
+ */
+int net_pkt_read_new(struct net_pkt *pkt, void *data, size_t length);
+
+/* Read u8_t data data a net_pkt */
+static inline int net_pkt_read_new_u8(struct net_pkt *pkt, u8_t *data)
```

Comment:
I suggest we change the name to `net_pkt_read_u8_new()` in order to be consistent. Same with the other value reader helpers.

Reply:
will be easier to remove the _new when necessary yes